src/libcrun/utils.c

```diff
@@ -941,7 +941,7 @@ is_current_process_confined (libcrun_error_t *err)
   if (UNLIKELY (bytes_read < 0))
     return crun_make_error (err, errno, "error reading file `%s`", attr_path);
 
-  return (strncmp (buf, "unconfined", bytes_read) != 0 && buf[0] != '\0');
+  return (strncmp (buf, "unconfined", 10) != 0);
```

Comment:
it can use unitialized data.

We need to enforce `bytes_read >= 10` and we could even use `memcmp` since we know the size, something like:

```
#define UNCONFINED "unconfined"
#define UNCONFINED_LEN (sizeof (UNCONFINED) - 1)
return bytes_read >= UNCONFINED_LEN && memcmp (buf, UNCONFINED, UNCONFINED_LEN));
```

Reply:
sounds much better :)
Thanks